pkg/virt-controller/watch/ovm.go

```diff
@@ -295,6 +283,64 @@ func (c *OVMController) startStop(ovm *virtv1.OfflineVirtualMachine, vm *virtv1.
 	return nil
 }
 
+// setupVMfromOVM creates a VirtualMachine object from one OfflineVirtualMachine object.
+func (c *OVMController) setupVMFromOVM(ovm *virtv1.OfflineVirtualMachine) *virtv1.VirtualMachine {
+	basename := c.getVirtualMachineBaseName(ovm)
+
+	vm := virtv1.NewVMReferenceFromNameWithNS(ovm.ObjectMeta.Namespace, "")
+	vm.ObjectMeta = ovm.Spec.Template.ObjectMeta
+	vm.ObjectMeta.Name = basename
+	vm.ObjectMeta.GenerateName = basename
+	vm.Spec = ovm.Spec.Template.Spec
+
+	setupStableFirmwareUUID(ovm, vm)
+
+	t := true
+	// TODO check if vm labels exist, and when make sure that they match. For now just override them
+	vm.ObjectMeta.Labels = ovm.Spec.Template.ObjectMeta.Labels
+	vm.ObjectMeta.OwnerReferences = []v1.OwnerReference{v1.OwnerReference{
+		APIVersion:         virtv1.OfflineVirtualMachineGroupVersionKind.GroupVersion().String(),
+		Kind:               virtv1.OfflineVirtualMachineGroupVersionKind.Kind,
+		Name:               ovm.ObjectMeta.Name,
+		UID:                ovm.ObjectMeta.UID,
+		Controller:         &t,
+		BlockOwnerDeletion: &t,
+	}}
+
+	return vm
+}
+
+// TODO: make it const?
+// no special meaning, randomly generated on my box.
+// TODO: do we want to use another constants? see examples in RFC4122
+var _uuid_ns = uuid.Parse("6a1a24a1-4061-4607-8bf4-a3963d0c5895")
+
+// setStableUUID makes sure the VM being started has a a 'stable' UUID.
+// The UUID is 'stable' if doesn't change across reboots.
+func setupStableFirmwareUUID(ovm *virtv1.OfflineVirtualMachine, vm *virtv1.VirtualMachine) {
+
+	logger := log.Log.Object(ovm)
+
+	if vm.Spec.Domain.Firmware == nil {
+		vm.Spec.Domain.Firmware = &virtv1.Firmware{}
+	}
+
+	existingUUID := vm.Spec.Domain.Firmware.UUID
+	if existingUUID != "" {
+		logger.Debugf("Using existing UUID '%s'", existingUUID)
+		return
+	}
+
+	stableUUID := types.UID(uuid.NewSHA1(_uuid_ns, []byte(vm.ObjectMeta.Name)).String())
+	if existingUUID == stableUUID {
```

Comment:
Hm, is this condition needed, as this is only `true` if ex and sta are empty ("").

Reply:
It is not needed, and if indeed the two UUIDs are equal, we just waste some cycles replacing the UUID with another identical. Maybe this `if` is more distracting then helpful, removing.